### ISSUE TITLE
`raw("\n")` doesn't render a newline

### ISSUE THREAD

**Issue by RubixDev** [typst/typst#2240]
### Description

```typ
// not monospace, but works as expected
#{"a";"\n";"b"}

// monospace, but the newline isn't rendered
#{raw("a");raw("\n");raw("b")}

// workaround: use normal text for newlines
#{raw("a");"\n";raw("b")}

// workaround: add a trailing zero-width space
#{raw("a");raw("\n" + sym.zws);raw("b")}

// workaround: use `linebreak()` instead
#{raw("a");linebreak();raw("b")}

// workaround: add an additional `\n`
#{raw("a");raw("\n\n");raw("b")}
```
![image](https://github.com/typst/typst/assets/35602040/8f4ef474-bd4e-4892-8efc-045bb2f3d9af)

Something just seems off, and the behavior is inconsistent between using `text()` and `raw()`. A call to `raw()` with any number of `\n`s always renders one newline less than expected. I also think this used to work in an older version of Typst, but I didn't double-check that and I may be remembering wrong.

Also note that using `raw("a\nb")` does what you expect.

### Operating system

Web app, Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Probably related to https://github.com/typst/typst/issues/597

**Comment by 8LWXpg** [comment]
after 0.10.0 update, this issue seems to be fixed somehow, here's some tests I made:

```typst
#set block(stroke: aqua+.5pt)

#block({raw("a");raw("\n");raw("b")})

#block({raw("a\n");raw("b\n")})

#block({raw("a\n");raw("b")})

#block({raw("a\nb")})

#block({raw("a\nb\n")})
```

<img width="40" alt="image" src="https://github.com/typst/typst/assets/105704427/aff6d2d5-85da-487b-96db-94785745d43c">

now only one trailing newlines at the end of consecutive text boxes get trimmed. This behavior seems acceptable as many coding styles suggest trailing newlines, it makes sense to trim one trailing newline with `#raw(read("file"))`

I can only guess this is caused by optimization that combines consecutive text boxes or something similar since I'm not familiar with any of the Typst's layout engin.

**Comment by RubixDev** [comment]
Should we close this then? I'd say it's working as intended now.

**Comment by laurmaedje** [comment]
Before closing, I'd like to know why it works now, to ensure it's really fixed in all cases.

**Comment by peng1999** [comment]
A bisect shows that this issue is fixed by d77356a16f3d2493709f5ebd78c0ce489a4fd98c.

**Comment by laurmaedje** [comment]
Looks like std's [`str::lines`](https://doc.rust-lang.org/std/primitive.str.html#method.lines) eats the final newline:

> The final line ending is optional. A string that ends with a final line ending will return the same lines as an otherwise identical string without a final line ending.
